powerstore/resource_host_group.go

```diff
@@ -420,6 +427,12 @@ func GetHostDetails(plan models.HostGroup, state *models.HostGroup) ([]string, [
 		removeHostIdsSlice = append(removeHostIdsSlice, hostID)
 	}
 
+	errmsg := ""
+	// check to disallow removing all the hosts from the host group when host connectivity is Local_Only
+	if plan.HostConnectivity.ValueString() == "Local_Only" && planHostIds == nil && removeHostIdsSlice != nil {
```

Comment:
so suppose id I have to destroy a host group without removing host. I will first have to set host connectivity to local only in one terraform apply, then remove host id in the second apply, and then perform destroy?
It seems to be a long process, is there any alternative for this?

Reply:
@shenda1 when we are removing all the hosts from host group, the host_connectivity becomes null and so we're getting state and plan mismatch error. So, we have added check which will verify if host connectivity in plan is set and if user is trying to remove all the hosts then, it will throw an error to remove the host_connectivity before deleting all the hosts.

Reply:
How are we expected to remove host connectivity? we can also set it to "Local_Only" for deletion

Reply:
we can remove hosts when host connectivity is "Local_Only" but after deletion it would be null so, we are asking user to remove it if it is "Local_Only" to avoid post-apply mismatch issue.

Reply:
How do we remove it? usually we set a field to "" (blank) in order to remove it, can we do the same here?

Reply:
setting blank is not possible as there are only 4 valid possible values for host_connectivity, we need to remove the attribute from config here